packages/codegen.go/src/helpers.ts

```diff
@@ -583,3 +583,84 @@ export function recursiveUnwrapMapSlice(item: go.PossibleType): go.PossibleType
 export function star(param: go.Parameter): string {
   return go.isRequiredParameter(param) || param.byValue ? '' : '*';
 }
+
+export type SerDeFormat = 'JSON' | 'XML';
+
+// used by getSerDeFormat to cache results
+const serDeFormatCache = new Map<string, SerDeFormat>();
+
+// returns the wire format for the named model.
+// at present this assumes the formats to be mutually exclusive.
+export function getSerDeFormat(model: go.ModelType | go.PolymorphicType, codeModel: go.CodeModel): SerDeFormat {
+  let serDeFormat = serDeFormatCache.get(model.name);
+  if (serDeFormat) {
+    return serDeFormat;
+  }
+
+  // for model-only builds we assume the format to be JSON
+  if (codeModel.clients.length === 0) {
```

Comment:
I don't think model-only builds for XML is a real thing but am following up.

Reply:
Should we throw if we do see them in XML?

Reply:
In swagger, one can define `consumes` and `produces` to define content types at the global level. For tsp, I don't believe there's such a construct; you can only define the content type via operations. This means that for model-only builds in tsp, we have no way to describe the wire format. So, there would be no way to detect this (see my email thread for more info on this).